plugins/tiledb/io/Bounds4D.hpp

```diff
@@ -0,0 +1,464 @@
+    };
+
+    using BOX3D::maxx;
+    using BOX3D::maxy;
+    using BOX3D::maxz;
+    using BOX3D::minx;
+    using BOX3D::miny;
+    using BOX3D::minz;
+    double mintm; ///< Minimum time value.
+    double maxtm; ///< Maximum time value.
+
+    BOX4D()
+        { clear(); }
+
+    BOX4D(const BOX4D& box) :
+    BOX3D(box), mintm(box.mintm), maxtm(box.maxtm)
+    {}
+
+    BOX4D& operator=(const BOX4D& box) = default;
+
+    explicit BOX4D(const BOX3D& box) :
+    BOX3D(box), mintm(0), maxtm(0) {}
+
+    explicit BOX4D(const BOX2D_& box) :
+    BOX3D(box.minx, box.miny, 0, box.maxx, box.maxy, 0), mintm(0), maxtm(0) {}
+
+    BOX4D(double minx, double miny, double minz, double mintm, double maxx,
+          double maxy, double maxz, double maxtm)
+          : BOX3D(minx, miny, minz, maxx, maxy, maxz), mintm(mintm), maxtm(maxtm)
+          {}
+
+    bool empty() const;
+
+    bool valid() const;
+
+    BOX4D& grow(double x, double y, double z, double tm);
+
+    void clear();
+
+    bool contains(double x, double y, double z, double tm) const
+    {
+        return BOX3D::contains(x, y, z) && mintm <= tm && tm <= maxtm;
+    }
+
+    bool contains(const BOX4D& other) const
+    {
+        return BOX3D::contains(other) &&
+        mintm <= other.mintm && other.maxtm <= maxtm;
+    }
+
+    bool equal(const BOX4D& other) const
+    {
+        return  BOX3D::equal(other) &&
+        mintm == other.mintm && maxtm == other.maxtm;
+    }
+
+    bool operator==(BOX4D const& rhs) const
+    {
+        return equal(rhs);
+    }
+
+    bool operator!=(BOX4D const& rhs) const
+    {
+        return (!equal(rhs));
+    }
+
+    BOX4D& grow(const BOX4D& other)
+    {
+        BOX3D::grow(other);
+        if (other.mintm < mintm) mintm = other.mintm;
+        if (other.maxtm > maxtm) maxtm = other.maxtm;
+        return *this;
+    }
+
+    BOX4D& grow(double dist)
+    {
+        BOX3D::grow(dist);
+        mintm -= dist;
+        maxtm += dist;
+        return *this;
+    }
+
+    void clip(const BOX4D& other)
+    {
+        BOX3D::clip(other);
+        if (other.mintm > mintm && other.mintm < maxtm) mintm = other.mintm;
+        if (other.maxtm < maxtm && other.maxtm > mintm) maxtm = other.maxtm;
+    }
+
+    bool overlaps(const BOX4D& other) const
+    {
+        return BOX3D::overlaps(other) &&
+        mintm <= other.maxtm && maxtm >= other.mintm;
+    }
+
+    BOX3D to3d() const
+    {
+        return *this;
+    }
+
+    BOX2D_ to2d() const
+    {
+        return this->to3d();
+    }
+
+    std::string toBox(uint32_t precision = 8) const
+    {
+        std::stringstream oss;
+
+        oss.precision(precision);
+        oss.setf(std::ios_base::fixed, std::ios_base::floatfield);
+
+        oss << "box4d(" << minx << " " << miny << " " << minz << " " << mintm << ", " <<
+        maxx << " " << maxy << " " << maxz << " " << maxtm << ")";
+        return oss.str();
+    }
+
+    std::string toWKT(uint32_t precision = 8) const
```

Comment:
Oh my, this is awful, isn't it? (Not your doing, just the nature of the beast.) Is it necessary? If it's just here for completeness maybe just drop it?